### DISCUSSION TITLE
Im trying to learn payload and I have been checking the examples for auth but im getting error when importing Access types

### DISCUSSION THREAD

**Discussion by GerardPolloRebozado** [payloadcms/payload#9981] [Q&A] [answered]
This code from the template doesnt works and both vscode and webstorm say "Cannot find module 'payload/config' or its corresponding type declarations."
```
import type { Access } from 'payload/config'

import { checkRole } from './checkRole'

const adminsAndUser: Access = ({ req: { user } }) => {
  if (user) {
    if (checkRole(['admin'], user)) {
      return true
    }

    return {
      id: user.id,
    }
  }

  return false
}

export default adminsAndUser
```

**Comment by zoku** [comment]
You should be able to import `Access` directly from `"payload"`-package (`import type {Access} from "payload"`.

**Reply by GerardPolloRebozado** [reply]
But the type from payload doesnt let you return the user id.

**Reply by andershermansen** [reply]
@GerardPolloRebozado Access type requires you to return either a boolean or a valid Where. You just need to adjust the return you are using a little bit:
```ts
    return {
      id: {
        equals: user.id,
      }
    }
```

**Reply by GerardPolloRebozado** [reply]
I'm not understanding what is this code is for. Where does the id comes from? user.id is the one that makes that request but what id is it comparing it to?

**Reply by andershermansen** [reply, marked as answer]
It can be used to apply access restrictions. So for example if you in your Users collection use this adminsAndUser on the read access defintion then any person with admin can see all users, but if you do not have admin then payload will apply the where (id equals user.id). The user object is the current logged in user, so when it's applied it will basically filter so that you only see your own user.

Read more about access control here: https://payloadcms.com/docs/access-control/overview

**Reply by GerardPolloRebozado** [reply]
ok thanks for the explanation. The examples should be updated so with the correct code

**Reply by andershermansen** [reply]
@GerardPolloRebozado Where is this example ?

**Reply by GerardPolloRebozado** [reply]
[Here](https://github.com/payloadcms/payload/blob/main/examples/auth/src/collections/access/adminsAndUser.ts)

**Reply by JoeCars** [reply]
I ran into the same issue, following the "auth" example and looking at the same file @GerardPolloRebozado mentioned. The change suggested fixes that error:

`import type { Access } from 'payload'`

But then the "adminsAndUser" variable throws a type error. Could the "auth" example be updated please, just trying to get a first project going.
